airflow/providers/postgres/datasets/postgres.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from urllib.parse import SplitResult
+
+
+def sanitize_uri(uri: SplitResult) -> SplitResult:
+    if not uri.netloc:
+        raise ValueError("URI format postgres:// must contain a host")
+    if uri.port is None:
+        host = uri.netloc.rstrip(":")
+        uri = uri._replace(netloc=f"{host}:5432")
+    path_parts = uri.path.split("/")
+    if len(path_parts) != 4:  # Leading slash, database, schema, and table names.
+        raise ValueError("URI format postgres:// must contain database, schema, and table names")
+    if not path_parts[2]:
+        path_parts[2] = "default"
```

Comment:
So this allows something like `postgres://host:5432/database//table`?

Reply:
Yes